include/gc/Dialect/Linalgx/Utils.h

```diff
@@ -0,0 +1,54 @@
+//===- Utils.h - linalgx utils ----------------------------------*- C++ -*-===//
+//
+// This file is licensed under the Apache License v2.0 with LLVM Exceptions.
+// See https://llvm.org/LICENSE.txt for license information.
+// SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+//
+//===----------------------------------------------------------------------===//
+
+#ifndef GC_DIALECTS_LINALGX_UTILS_H
+#define GC_DIALECTS_LINALGX_UTILS_H
+
+#include "mlir/Dialect/Linalg/IR/Linalg.h"
+#include "mlir/IR/BuiltinTypes.h"
+#include "mlir/IR/TypeUtilities.h"
+#include "mlir/IR/Types.h"
+
+namespace mlir {
+namespace linalgx {
+
+/// @brief enum of type of matmul packing
+enum class PackingType {
+  MM4D = 0,    // MKmk x NKkn
+  VNNI_MM2D,   // MK x NKknV
+  VNNI_MM4D,   // MKmk x NKknV
+  VNNI_BRMM3D, // BMK x BKNV
+  NUM_TYPES,
+};
+
+/// @brief make a generic packed matmul Op based on PackingType
+/// @param builder builder
+/// @param loc location
+/// @param opType the PackingType
+/// @param inputs matmul A, B
+/// @param outputs matmul C
+/// @return the generic packed matmul Op
+FailureOr<linalg::GenericOp>
+makeGenericPackedMatmulOp(OpBuilder &builder, Location loc, PackingType opType,
+                          ValueRange inputs, ValueRange outputs);
+
+/// @brief identify a generic packed matmul Op based on PackingType
+/// @param op the op
+/// @param opType the PackingType
+/// @return true if op is a generic packed matmul Op
+bool isGenericPackedMatmulOp(Operation *op, PackingType opType);
+
+/// @brief identify a matmul Op based on ContractionOp and PackingType
+/// @param op the op
+/// @return true if op is a matmul Op
+bool isMatmulOp(Operation *op);
```

Comment:
Added `isMatmulOp` interface to check ContractionOp and all PackingType. Maybe there is a more structured way to check for packing matmuls that can replace `NotProjectedPermutations`.